addons/web_editor/static/src/js/editor/rte.summernote.js

```diff
@@ -1090,44 +1090,24 @@ var SummernoteManager = Class.extend(mixins.EventDispatcherMixin, ServicesMixin,
      */
     saveCroppedImages: function ($editable) {
         var defs = _.map($editable.find('.o_cropped_img_to_save'), async croppedImg => {
-            var $croppedImg = $(croppedImg);
-            $croppedImg.removeClass('o_cropped_img_to_save');
-
-            var resModel = $croppedImg.data('crop:resModel');
-            var resID = $croppedImg.data('crop:resID');
-            var cropID = $croppedImg.data('crop:id');
-            var mimetype = $croppedImg.data('crop:mimetype');
-            var originalSrc = $croppedImg.data('crop:originalSrc');
-
-            var datas = $croppedImg.attr('src').split(',')[1];
-            let attachmentID = cropID;
-            if (!cropID) {
-                var name = originalSrc + '.crop';
-                attachmentID = await this._rpc({
-                    model: 'ir.attachment',
-                    method: 'create',
-                    args: [{
-                        res_model: resModel,
-                        res_id: resID,
-                        name: name,
-                        datas: datas,
-                        mimetype: mimetype,
-                        url: originalSrc, // To save the original image that was cropped
-                    }],
-                });
-            } else {
-                await this._rpc({
-                    model: 'ir.attachment',
-                    method: 'write',
-                    args: [[cropID], {datas: datas}],
-                });
-            }
-            const access_token = await this._rpc({
-                model: 'ir.attachment',
-                method: 'generate_access_token',
-                args: [[attachmentID]],
+            croppedImg.classList.remove('o_cropped_img_to_save');
+
+            const croppedAttachmentSrc = await this._rpc({
+                route: '/web_editor/crop_attachment',
+                params: {
+                    id: parseInt(croppedImg.dataset.attachmentId),
+                    res_model: croppedImg.dataset.resModel,
+                    res_id: parseInt(croppedImg.dataset.resId),
+                    name: croppedImg.dataset.originalName + '.crop',
+                    data: croppedImg.getAttribute('src').split(',')[1],
+                    mimetype: croppedImg.dataset.mimetype,
+                    original_id: parseInt(croppedImg.dataset.originalId),
+                },
+            });
+            croppedImg.setAttribute('src', croppedAttachmentSrc);
+            weWidgets.ImageCropWidget.prototype.removeOnSaveAttributes.forEach(attr => {
+                delete croppedImg.dataset[attr];
             });
-            $croppedImg.attr('src', '/web/image/' + attachmentID + '?access_token=' + access_token[0]);
```

Comment:
It's now all done in the controller crop_attachment route on the server, so that I can avoid an RPC round-trip to first create the attachment and then query the db for the image_src.

Reply:
I understood that :wink: What I don't see is the access_token generation server-side?

Reply:
For some reason I thought that _compute_image_src did that but it does not. I guess I confused access_token and unique. Added something to handle that in the controller as well.